tracing-core/src/dispatcher.rs

```diff
@@ -168,6 +171,20 @@ const INITIALIZING: usize = 1;
 const INITIALIZED: usize = 2;
 
 static mut GLOBAL_DISPATCH: Option<Dispatch> = None;
+// The global default dispatch that should be used by [`Enter::with_current`]
+// when there is no explicit local dispatch set.
+//
+// By default this is initialized to [`NO_DISPATCH`] and will be set to
+// [`GLOBAL_DISPATCH`] when this is being set.
+static GLOBAL_DEFAULT_DISPATCH: Lazy<AtomicPtr<Dispatch>> =
+    Lazy::new(|| AtomicPtr::new(unsafe { &mut *NO_DISPATCH as *mut _ }));
+
+// This is only `mut` to make it work as default value of [`GLOBAL_DEFAULT_DISPATCH`].
+//
+// We can not use `new` as we don't want this to be registered at the callsites.
+static mut NO_DISPATCH: Lazy<Dispatch> = Lazy::new(|| Dispatch {
+    subscriber: Arc::new(NoSubscriber::default()),
+});
```

Comment:
it occurs to me that we could potentially avoid the need to have two `Lazy` cells and a `static mut` if we made a way for a "no dispatcher" `Dispatch` to be constructed in a `const fn`. we could investigate changing `Dispatch` to:
```rust
pub struct Dispatch {
    subscriber: Option<Arc<dyn Subscriber + ...>>,
}
```
and having the `None` variant always nop. Of course, we'd want to ensure *that* doesn't have a worse performance impact, but it might be worth considering.

It would also make cloning the `NO_DISPATCH` dispatcher cheaper, as it wouldn't have to do an `Arc` refcount bump...